doc/releases/release-notes-4.1.rst

```diff
@@ -33,6 +33,29 @@ We are pleased to announce the release of Zephyr version 4.1.0.
 
 Major enhancements with this release include:
 
+**Performance improvements**
+  Multiple performance improvements of core Zephyr kernel functions have been implemented,
+  benefiting all supported hardware architectures.
+
+  An official port of the :zephyr_file:`thread_metric <tests/benchmarks/thread_metric>` RTOS
+  benchmark has also been added to make it easier for developers to measure the performance of
+  Zephyr on their hardware and compare it to other RTOSes.
+
+**Experimental support for IAR compiler**
+  :ref:`toolchain_iar_arm` can now be used to build Zephyr applications. This is an experimental
+  feature that is expected to be improved in future releases.
+
+**Initial support for Rust on Zephyr**
+  <placeholder>
+
+**USB MIDI Class Driver**
+  Introduction of a new :ref:`USB MIDI 2.0 <usbd_midi2>` device driver, allowing Zephyr devices to
+  communicate with MIDI controllers and instruments over USB.
+
+**Expanded Board Support**
+  Support for 70 :ref:`new boards <boards_added_in_zephyr_4_1>` and 11
+  :ref:`new shields <shields_added_in_zephyr_4_1>` has been added in this release.
+
```

Comment:
The batch of usb-canbus boards and https://github.com/CANnectivity/?

Reply:
Yes!

Reply:
Also would probably want to highlight Pico 2 and WCH -- maybe feels odd to put the spotlights on those two in particular but they are both significant community contributions worth highlighting IMO. Maybe a couple other boards deserve similar shoutout?

Reply:
yeah it is odd but then those two in particular stands out for the combination of cheap + community that made them very popular in hobbyst community, the only other one in that class are the various esp32 modules but I don't think there was anything new on that front in this cycle

Reply:
addressed, I think - please check!